catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SortedQueryMonitor.java

```diff
@@ -191,22 +199,22 @@ public void run() {
     }
 
     private void timeoutRemainingSources(Set<ProcessingDetails> processingDetails) {
-        for (Source expiredSource : futures.values()) {
+        for (QueryRequest expiredSource : futures.values()) {
             if (expiredSource != null) {
-                LOGGER.info("Search timed out for {}", expiredSource.getId());
-                processingDetails.add(new ProcessingDetailsImpl(expiredSource.getId(),
+                LOGGER.info("Search timed out for {}", getSourceIdFromRequest(expiredSource));
+                processingDetails.add(new ProcessingDetailsImpl(getSourceIdFromRequest(expiredSource),
                         new TimeoutException()));
             }
```

Comment:
If you want to reduce branch complexity, you could do something like:

futures.value().stream().filter(Objects::notNull).foreach( ... )